core/node/state_keeper/src/mempool_actor.rs

```diff
@@ -89,20 +89,35 @@ impl MempoolFetcher {
                 .await
                 .context("failed getting pending protocol version")?;
 
-            let l2_tx_filter = l2_tx_filter(
-                self.batch_fee_input_provider.as_ref(),
-                protocol_version.into(),
-            )
-            .await
-            .context("failed creating L2 transaction filter")?;
+            let (fee_per_gas, gas_per_pubdata) = if let Some(unsealed_batch) = storage
+                .blocks_dal()
+                .get_unsealed_l1_batch()
+                .await
+                .context("failed getting unsealed batch")?
+            {
+                let (fee_per_gas, gas_per_pubdata) = derive_base_fee_and_gas_per_pubdata(
+                    unsealed_batch.fee_input,
+                    protocol_version.into(),
+                );
+                (fee_per_gas, gas_per_pubdata as u32)
+            } else {
+                let filter = l2_tx_filter(
+                    self.batch_fee_input_provider.as_ref(),
+                    protocol_version.into(),
+                )
+                .await
+                .context("failed creating L2 transaction filter")?;
```

Comment:
Why do we even create a filter here? Can't we do something like this instead:
```rust
let fee_input = if let Some(unsealed) = ... {
    unsealed_batch.fee_input
} else {
    batch_fee_input_provider.get_batch_fee_input().await?
}
let (fee_per_gas, gas_per_pubdata) = derive_base_fee_and_gas_per_pubdata(
    unsealed_batch.fee_input,
    protocol_version.into(),
);
```

Reply:
Also, looking at it again, aren't you doing pretty the same as what https://github.com/matter-labs/zksync-era/pull/2868 is doing just on a smaller scale? Or do you intend this to be a temporary fix specifically for mempool?

Reply:
> Why do we even create a filter here?

I'm creating filter so mempool_actor and mempool_io are using the same `l2_tx_filter` method, so there are less chances that someone changes it only for one of them.

It's connected to 2868 but separate: your PR solves that API may return incorrect fee params, my PR solves that mempool actor may use incorrect fee params

Reply:
Ah, yes, it's using a different `BatchFeeModelInputProvider` impl underneath, makes sense.